pkg/app/piped/notifier/matcher_test.go

```diff
@@ -148,6 +148,14 @@ func TestMatch(t *testing.T) {
 						},
 					},
 				}: true,
+				{
+					Type: model.NotificationEventType_EVENT_DEPLOYMENT_STARTED,
+					Metadata: &model.NotificationEventDeploymentStarted{
+						Deployment: &model.Deployment{
+							ApplicationName: "canary",
+						},
+					},
+				}: true,
```

Comment:
Could you please add a test case to return false?

```suggestion
				{
					Type: model.NotificationEventType_EVENT_DEPLOYMENT_STARTED,
					Metadata: &model.NotificationEventDeploymentStarted{
						Deployment: &model.Deployment{
							ApplicationName: "canary",
						},
					},
				}: true,
				{
					Type: model.NotificationEventType_EVENT_DEPLOYMENT_STARTED,
					Metadata: &model.NotificationEventDeploymentStarted{
						Deployment: &model.Deployment{
							ApplicationName: "bluegreen",
						},
					},
				}: false,
```

Reply:
Sure, added in [`edfb792` (#5523)](https://github.com/pipe-cd/pipecd/pull/5523/commits/edfb792bb57af81679f5b75f7bc0b74441e89616).